File: chroma-league/src/test/java/net/booone/chromaleague/hud/parts/resource/AbstractResourceTest.java
package net.booone.chromaleague.hud.parts.resource;

import net.booone.chromaleague.GameStateMocks;
import net.booone.chromaleague.rest.activeplayer.ChampionStats;
import org.junit.jupiter.api.BeforeEach;

import static org.mockito.Mockito.when;

public abstract class AbstractResourceTest {
    protected GameStateMocks gameStateMocks;

    @BeforeEach
    void setUp() {
        gameStateMocks = new GameStateMocks();
    }

    protected void mockResource(double resource, double maxResource) {
        final ChampionStats mockedChampionStats = gameStateMocks.championStats();
        when(mockedChampionStats.resourceValue()).thenReturn(resource);
        when(mockedChampionStats.resourceMax()).thenReturn(maxResource);
    }

    protected void mockRange(double range) {
        final ChampionStats mockedChampionStats = gameStateMocks.championStats();
        when(mockedChampionStats.attackRange()).thenReturn(range);
    }
}

File: test/hud/parts/resource/AbstractResourceTest.ts
import {beforeEach} from 'vitest';
import {GameStateMocks} from '../../../GameStateMocks.js';
import {ChampionStats} from '../../../../src/rest/activeplayer/ChampionStats.js';

export let gameStateMocks: GameStateMocks;

export function setupResourceTest(): void {
    beforeEach(() => {
        gameStateMocks = new GameStateMocks();
    });
}

export function mockResource(resource: number, maxResource: number): void {
    const stats: ChampionStats = gameStateMocks.championStats();
    stats.resourceValue = resource;
    stats.resourceMax = maxResource;
}

export function mockRange(range: number): void {
    const stats: ChampionStats = gameStateMocks.championStats();
    stats.attackRange = range;
}
